rust/src/types/memory.rs

```diff
@@ -129,7 +129,63 @@ mod tests {
     }
 
     #[test]
-    fn get_slice() {
+    fn expand() {
+        let mut memory = Memory::new(Vec::new());
+        assert_eq!(memory.expand(1, &mut 1_000), Ok(()));
+        assert_eq!(memory.into_inner(), [0; 32]);
+
+        let mut memory = Memory::new(Vec::new());
+        assert_eq!(memory.expand(32, &mut 1_000), Ok(()));
+        assert_eq!(memory.into_inner(), [0; 32]);
+
+        let mut memory = Memory::new(vec![1; 32]);
+        assert_eq!(memory.expand(64, &mut 1_000), Ok(()));
+        assert_eq!(memory.into_inner(), {
+            let mut mem = [1; 64];
+            mem[32..].copy_from_slice(&[0; 32]);
+            mem
+        });
+
+        let mut memory = Memory::new(Vec::new());
+        assert_eq!(
+            memory.expand(u64::MAX, &mut 1_000),
+            Err(StatusCode::EVMC_OUT_OF_GAS)
+        );
+    }
+
+    #[test]
+    fn consume_expansion_cost() {
+        let memory = Memory::new(Vec::new());
+        let mut gas_left = 0;
+        assert_eq!(memory.consume_expansion_cost(&mut gas_left, 0), Ok(()));
```

Comment:
`consume_expansion_cost` did not change order, still has `(gas_left, cost)` format, the other functions have inverted them,

Reply:
The second parameter is in this case not the cost but the length to which the memory should be expanded. But I agree that the order should be switched.